framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -435,22 +443,24 @@ private AnnotationMirror createBooleanAnnotationMirror(List<Boolean> values) {
 
     private List<Number> calculateNumericalUnaryOp(
             Node operand, NumericalUnaryOps op, TransferInput<CFValue, CFStore> p) {
-        List<? extends Number> lefts = getNumericalValues(operand, p);
         List<Number> resultValues = new ArrayList<>();
-        for (Number left : lefts) {
-            NumberMath<?> nmLeft = NumberMath.getNumberMath(left);
-            switch (op) {
-                case PLUS:
-                    resultValues.add(nmLeft.unaryPlus());
-                    break;
-                case MINUS:
-                    resultValues.add(nmLeft.unaryMinus());
-                    break;
-                case BITWISE_COMPLEMENT:
-                    resultValues.add(nmLeft.bitwiseComplement());
-                    break;
-                default:
-                    throw new UnsupportedOperationException();
+        if (!isIntRange(operand, p)) {
```

Comment:
The control flow type refinement is a nice touch; I don't see why to remove it since you have implemented it.  Thanks!